libraries/AP_GPS/AP_GPS_NOVA.cpp

```diff
@@ -257,6 +278,42 @@ AP_GPS_NOVA::process_message(void)
         _new_position = true;
     }
 
+    // 971 for unicore 2042 for novatel
+    if(messageid == 971 || messageid == 2042)
```

Comment:
I think you need to actually only enter this if it's unicore and 971 or, novatel and 2042.